project_structure.txt

```diff
@@ -0,0 +1,119 @@
+.modin/
+   .config/
+   .distributed/
+   .pandas/
+   .test/
+   .rest files
+   .core/
+      .io/
+         .column_stores/
+            . # the same as it is now
+         .sql/
+            . # the same as it is now
+         .text/
+            . # the same as it is now
+         .file_dispatcher.py # the same as it is now
+         .io.py # the same as it is now
+      .dispatching/
+         .factories/
+            . # the same as it is now
+      .formats/
+         .base/
+            .query_compiler.py # BaseQueryCompiler
+         .pandas/
+            .base/ # or default (DefaultPandasQueryCompiler)
+               .query_compiler.py # BasePandasQueryCompiler
+            .kernels.py
+            .query_compiler.py # PandasQueryCompiler
+            .parsers.py
+            .utils.py # data_management/utils moved here
+      .execution/
+        .ray/
+           .frame/
+              .dataframe.py # GenericRayDataframe
+              .metadata.py # GenericRayDataframeMetadata
+           .partition_management/ # or `partitioning`
+              .axis_partition.py # GenericRayDataframeAxisPartition
+              .partition_manager.py # GenericRayDataframeManager
+              .partition.py # GenericRayDataframePartition
+           .io/
+              .io.py # GenericRayIO
+           .task_wrapper.py # GenericRayTask
+           .utils.py # init_ray
+        .dask/
+           . # the same as for ray
+        .python/
+           . # the same as for ray
+      .dataframe/  # implementations go here
+         .algebra/
+            .operators/ # dispatching/functions here
+               . # the same as it is in dispatching/functions
+         .implementations/ # all of the implementations go here
+            .base/
+               .frame/
+                  .dataframe.py # BaseDataframe
+                  .metadata.py # BaseDataframeMetaData
+               .partitioning/
+                  .axis_partition.py # BaseDataframeAxisPartition
+                  .partition_manager.py # BaseDataframeManager
+                  .partition.py # BaseDataframePartition
+            .pandas_base/ # make distinct from pandas format
```

Comment:
```suggestion
            .pandas/ # make distinct from pandas format
```
I think `_base` part is redundant.

Reply:
What do you think if we put pandas_on_ray, etc. under pandas folder?

Reply:
I would like all of the implementations to be top-level so we can see what all is supported from one place.

I don't like keeping the name as just `pandas` because it could be confused with the `storage_format` above.

Reply:
Don't you think it can become a bit bloated? I mean `pandas_on_ray`, `pandas_on_dask`, `pandas_on_python`, `pyarrow_on_ray`, `pyarrow_on_dask`, `pyarrow_on_python`, and future implementations will be located in the place. However, it would look gracefully with the Scaleout API. `pandas_on_scaleout`, `pyarrow_on_scaleout`, etc. :)

As to `_base`, I don't understand why it could be confused with the `storage_format` above because they are located in different folders (`storage_formats` and `implementations`). Could you elaborate?